slsDetectorSoftware/src/Module.cpp

```diff
@@ -1746,19 +1747,109 @@ void Module::setLEDEnable(bool enable) {
 // Pattern
 
 void Module::setPattern(const std::string &fname) {
-    uint64_t word;
-    uint64_t addr = 0;
-    FILE *fd = fopen(fname.c_str(), "r");
-    if (fd != nullptr) {
-        while (fread(&word, sizeof(word), 1, fd) != 0U) {
-            setPatternWord(addr, word); // TODO! (Erik) do we need to send
-                                        // pattern in 64bit chunks?
-            ++addr;
+    patternParameters pat;
+    memset(&pat, 0, sizeof(pat));
+    std::ifstream input_file;
+    input_file.open(fname.c_str(), std::ios_base::in);
+    if (!input_file.is_open()) {
+        throw RuntimeError("Could not open pattern file " + fname +
+                           " for reading");
+    }
+    std::string current_line;
+    while (input_file.good()) {
+        getline(input_file, current_line);
+        if (current_line.find('#') != std::string::npos) {
+            current_line.erase(current_line.find('#'));
+        }
+        LOG(logDEBUG1) << "current_line after removing comments:\n\t"
+                       << current_line;
+        if (current_line.length() > 1) {
+
+            // convert command and string to a vector
+            std::istringstream iss(current_line);
```

Comment:
It might be possible to use the iss directly without allocating a vector of strings. But this is something we can look into later. Anyway I don't think this is a performance critical part of the software.